src/events/mod.rs

```diff
@@ -353,7 +353,7 @@ impl<'a> BytesStart<'a> {
         &'a self,
         attr_name: N,
     ) -> Result<Option<Attribute<'a>>> {
-        for a in self.attributes() {
+        for a in self.attributes().with_checks(false) {
```

Comment:
My thinking here is, the code already stops at the first instance of the correct key and returns early, so there's not much point doing checks.  And if you're asking for one specific element you probably don't care so much that others might be in violation of the rules.

This was about 50% faster on top of the other improvement.